v4/feature/trade/src/main/java/exchange/dydx/trading/feature/trade/tradeinput/DydxTradeInputMarginModeViewModel.kt

```diff
@@ -37,25 +41,33 @@ class DydxTradeInputMarginModeViewModel @Inject constructor(
             title = localizer.localize("APP.GENERAL.MARGIN_MODE"),
             asset = tradeInput?.marketId ?: "",
             crossMargin = DydxTradeInputMarginModeView.MarginTypeSelection(
-                localizer.localize("APP.GENERAL.CROSS_MARGIN"),
-                localizer.localize("APP.GENERAL.CROSS_MARGIN_DESCRIPTION"),
-                marginMode == MarginMode.cross,
+                title = localizer.localize("APP.GENERAL.CROSS_MARGIN"),
+                text = localizer.localize("APP.GENERAL.CROSS_MARGIN_DESCRIPTION"),
+                selected = marginMode == MarginMode.cross,
             ) {
                 abacusStateManager.trade("CROSS", TradeInputField.marginMode)
-                router.navigateBack()
+                closeView()
             },
             isolatedMargin = DydxTradeInputMarginModeView.MarginTypeSelection(
-                localizer.localize("APP.GENERAL.ISOLATED_MARGIN"),
-                localizer.localize("APP.GENERAL.ISOLATED_MARGIN_DESCRIPTION"),
-                marginMode == MarginMode.isolated,
+                title = localizer.localize("APP.GENERAL.ISOLATED_MARGIN"),
+                text = localizer.localize("APP.GENERAL.ISOLATED_MARGIN_DESCRIPTION"),
+                selected = marginMode == MarginMode.isolated,
             ) {
                 abacusStateManager.trade("ISOLATED", TradeInputField.marginMode)
-                router.navigateBack()
+                closeView()
             },
             errorText = null,
             closeAction = {
-                router.navigateBack()
+                closeView()
             },
         )
     }
+
+    private fun closeView() {
+        router.navigateBack()
+        appScope.launch {
+            delay(1.seconds) // Delay to allow the back navigation to complete
```

Comment:
1 second is quite long... shouldn't we know how long the animation is based on where ever we are configuring it?

Reply:
Delay is a temporary solution for now.  We will need to properly fix it before releasing

Reply:
kk, let's make a ticket.